broker/config.go

```diff
@@ -0,0 +1,170 @@
+	Username                     string                    `json:"username"`
+	Password                     string                    `json:"password"`
+	Region                       string                    `json:"region"`
+	ClusterNamePrefix            string                    `json:"cluster_name_prefix"`
+	BrokerName                   string                    `json:"broker_name"`
+	AuthTokenSeed                string                    `json:"auth_token_seed"`
+	AllowUserProvisionParameters bool                      `json:"allow_user_provision_parameters"`
+	AllowUserUpdateParameters    bool                      `json:"allow_user_update_parameters"`
+	AllowUserBindParameters      bool                      `json:"allow_user_bind_parameters"`
+	CacheSubnetGroupName         string                    `json:"cache_subnet_group_name"`
+	VpcSecurityGroupIds          []string                  `json:"vpc_security_group_ids"`
+	Catalog                      brokerapi.CatalogResponse `json:"catalog"`
+	PlanConfigs                  map[string]PlanConfig     `json:"plan_configs"`
+}
+
+func (cfg *Config) GetPlanConfig(planID string) (PlanConfig, error) {
+	plan, ok := cfg.PlanConfigs[planID]
+	if !ok {
+		return PlanConfig{}, ErrNoSuchPlan
+	}
+	return plan, nil
+}
+
+func LoadConfig(configFile string) (config Config, err error) {
+	if configFile == "" {
+		return config, errors.New("Must provide a config file")
+	}
+
+	file, err := os.Open(configFile)
+	if err != nil {
+		return config, err
+	}
+	defer file.Close()
+
+	bytes, err := ioutil.ReadAll(file)
+	if err != nil {
+		return config, err
+	}
+
+	if err = json.Unmarshal(bytes, &config); err != nil {
+		return config, err
+	}
+
+	if err = config.Validate(); err != nil {
+		return config, fmt.Errorf("Validating config contents: %s", err)
+	}
+
+	return config, nil
+}
+
+func (c Config) Validate() error {
+	if c.LogLevel == "" {
+		return errors.New("Must provide a non-empty log_level")
+	}
+
+	if c.Username == "" {
+		return errors.New("Must provide a non-empty username")
+	}
+
+	if c.Password == "" {
+		return errors.New("Must provide a non-empty password")
+	}
+
+	if c.Region == "" {
+		return errors.New("Must provide a non-empty region")
+	}
+
+	if c.ClusterNamePrefix == "" {
+		return errors.New("Must provide a non-empty cluster_name_prefix")
+	}
+
+	if c.BrokerName == "" {
+		return errors.New("Must provide a non-empty broker_name")
+	}
+
+	if c.AuthTokenSeed == "" {
+		return errors.New("Must provide a non-empty auth_token_seed")
+	}
+
+	if c.CacheSubnetGroupName == "" {
+		return errors.New("Must provide a cache_subnet_group_name")
+	}
+
+	if len(c.VpcSecurityGroupIds) < 1 {
+		return errors.New("Must provide at least one VPC security group ID")
+	}
+
+	for _, s := range c.Catalog.Services {
+		for _, p := range s.Plans {
+			if !c.hasPlanConfig(p.ID) {
+				return fmt.Errorf("Plan with ID %s has no PlanConfig", p.ID)
+			}
+		}
+	}
+
+	for k, _ := range c.PlanConfigs {
```

Comment:
FYI, you don't need to provide the second variable (in this case `_`) if you're going to ignore it ;)

Reply:
🙌